disperser/dataapi/subgraph_client.go

```diff
@@ -262,39 +269,79 @@ func getOperatorsWithRegisteredDeregisteredIntervalEvents(
 		}
 	}
 
+	currentTs := uint64(time.Now().Unix())
 	operatorEventsChan := make(chan OperatorEvents, 1)
 	go func() {
 		defer close(operatorEventsChan)
-		pool := workerpool.New(10)
-		for o, e := range operators {
-			operatorId, events := o, e
-			pool.Submit(func() {
-				var lastDeregistered uint64
-				for i := 0; i < len(events[0]); i++ {
-					var (
-						// we start with the assumption that the operator is still registered
-						// a 0 value means that the operator is still registered
-						deregistered uint64
-						registered   = events[0][i]
-					)
-
-					// if there is a deregistered event that macthes in same position of registered event, we use it
-					if i < len(events[1]) {
-						deregistered = events[1][i]
-					}
 
-					// if there is no deregistered event that matches in same position of registered event, we use the last one
-					if i == 0 || registered > lastDeregistered {
-						lastDeregistered = deregistered
-						operatorEventsChan <- OperatorEvents{
-							OperatorId: operatorId,
-							Events:     []uint64{registered, deregistered},
+		// For the time window [blockTimestamp, now], compute the sub intervals during
+		// which the operator is live in EigenDA network for validating batches.
+		for operatorId := range nonSigners {
+			reg := operators[operatorId][0]
+			dereg := operators[operatorId][1]
+			if len(reg) == 0 && len(dereg) == 0 {
+				// The operator has no registration/deregistration events: it's live
+				// for the entire time window.
+				operatorEventsChan <- OperatorEvents{
+					OperatorId: operatorId,
+					Events:     []uint64{blockTimestamp, currentTs},
+				}
+			} else if len(reg) == 0 {
+				// The operator has only deregistration event: it's live from the beginning
+				// of the time window until the deregistration.
+				operatorEventsChan <- OperatorEvents{
+					OperatorId: operatorId,
+					Events:     []uint64{blockTimestamp, dereg[0]},
+				}
+			} else if len(dereg) == 0 {
+				// The operator has only registration event: it's live from registration to
+				// the end of the time window.
+				operatorEventsChan <- OperatorEvents{
+					OperatorId: operatorId,
+					Events:     []uint64{reg[0], currentTs},
+				}
+			} else {
+				// The operator has both registration and deregistration events in the time
+				// window.
+				if reg[0] < dereg[0] {
+					// The first event in the time window is registration. This means at
+					// the beginning (i.e. blockTimestamp) it's not live.
+					for i := 0; i < len(reg); i++ {
+						if i < len(dereg) {
```

Comment:
we have len(reg) = len(dereg) or len(reg) = len(dereg) +1 here?

shouldn't this be i <= len(dereg)

Reply:
Yeah, I think we will have this pattern of logs on the timeline: `reg-dereg-reg-dereg-....`
- The very first event will be registration
- The registration / deregistration log events will alternate

This is because 1) register an operator that's already registered will fail; 2) deregister an operator that's already deregistered will fail.

Reply:
my bad, you're right